### PR TITLE
fix(api): restore SSE Content-Type for streaming responses

### PR DESCRIPTION
**Pull request by pescn**
## Summary

- All three streaming endpoints (`/v1/chat/completions`, `/v1/responses`, `/v1/messages`) were returning `Content-Type: text/plain` instead of `text/event-stream`. Bodies were correctly SSE-formatted but the header lied, breaking strict SSE clients (browser `EventSource`, EvalScope perf benchmark, any RFC 8941 implementation).
- Switch each streaming branch to return a native `Response` wrapping a `ReadableStream`. Elysia detects pre-formatted Responses and runs `handleStream` with `skipFormat=true`, preserving our headers and bypassing 1.4's auto SSE wrapping.
- Add `Accept: text/event-stream` content negotiation: when `body.stream` is unset, an `Accept` header listing `text/event-stream` enables streaming. Body always wins when explicit.

## Root cause

When deps were bumped from `elysia ^1.3.5` to `^1.4.16` ([d1f186a](https://github.com/EM-GeekLab/NexusGate/commit/d1f186a)), the default `Content-Type` for async-generator returns silently changed from `text/event-stream` to `text/plain`. In 1.4 the auto SSE path kicks in only when `set.headers[\"content-type\"]` (lowercase) starts with `text/event-stream`, **and** it then wraps every yielded chunk with `data: \${chunk}\n\n`. Our adapters already emit pre-formatted SSE strings, so opting in to that auto path would double-prefix `data:`. Setting `set.headers[\"Content-Type\"]` (mixed case) leaves both keys live and Bun joins them as `text/event-stream, text/plain`. Returning a native `Response` is the only path in 1.4 that lets us set the header without triggering double-wrapping.

Verified against three historical versions:

| Version | Date | Elysia | `stream:true` Content-Type |
|---|---|---|---|
| `eebe5a2` | 2025-09-09 | `^1.3.5` | `text/event-stream` ✓ |
| `3223b76` | 2025-11-20 | `^1.4.16` | `text/plain` ❌ |
| `main` (this PR) | now | `^1.4.22` | `text/event-stream; charset=utf-8` ✓ |

## Test plan

- [x] `curl -D -` against `/v1/chat/completions` with `stream:true` → `Content-Type: text/event-stream; charset=utf-8`, body is valid SSE ending in `data: [DONE]\n\n`
- [x] Same for `/v1/messages` (Anthropic) → ends with `event: message_stop`
- [x] Same for `/v1/responses` (OpenAI Responses API) → events stream correctly
- [x] Non-streaming branches still return `Content-Type: application/json` (regression check)
- [x] `Accept: text/event-stream` (no `body.stream`) → streaming
- [x] `Accept: text/event-stream` + `stream:false` → JSON (body wins)
- [x] Weighted Accept list (`text/event-stream;q=1, application/json;q=0.5`) → streaming
- [x] `bun run lint` and `bun run check` pass

Closes #81

## 发行说明

* **新功能**
  * 自动根据客户端 Accept 头启用事件流（SSE）传输。
  * 新增用于识别是否接受 SSE 的解析工具函数。
* **改进**
  * 流式响应改为原生 SSE 响应，显式设置 Content-Type/连接/缓存头，提升稳定性与兼容性。
  * 流内错误改为通过 SSE 事件发送并记录，改善错误可见性与连接处理。
* **测试**
  * 新增单元测试覆盖 Accept 头解析与相关行为。

### CODE REVIEW

File: backend/src/utils/api-helpers.ts [hunk 1]
```diff
@@ -46,6 +46,21 @@ export const EXCLUDED_HEADERS = new Set([
...
+ * weighted lists like `text/event-stream;q=1, application/json;q=0.5`.
+ */
+export function acceptsEventStream(headers: Headers): boolean {
+  const accept = headers.get("accept");
+  if (!accept) {
+    return false;
+  }
+  return accept
+    .split(",")
+    .some((part) => part.trim().toLowerCase().startsWith("text/event-stream"));
```
[the remark this answers is not in the document]
Comment (pescn) [reply backend/src/utils/api-helpers.ts]: Fixed in f30fe7a — both points were valid and confirmed reproducible against the prior implementation:

- `text/event-stream;q=0` and `text/event-stream;q=0, application/json` previously returned `true`
- `text/event-stream+json` previously returned `true`

The new helper splits each comma-separated entry on `;`, requires the media type to be exactly `text/event-stream` (lowercased), and parses the `q` parameter — entries with `q=0` (or any non-positive / non-finite q) are filtered out. Default q is 1 when absent, matching RFC 7231 §5.3.1.

Regression tests added in `backend/src/utils/api-helpers.test.ts` (12 cases covering q=0, q=0.0, structured suffixes, weighted lists, case/whitespace variants, and the malformed-q edge case).

File: backend/src/utils/api-helpers.ts [hunk 2]
```diff
@@ -46,6 +46,21 @@ export const EXCLUDED_HEADERS = new Set([
...
+}
```
[the remark this answers is not in the document]
Comment (pescn) [reply backend/src/utils/api-helpers.ts:82]: Fixed in f30fe7a. The helper now parses the `q` parameter from each comma-separated entry and filters out non-positive values, so `Accept: text/event-stream;q=0` is correctly treated as "do not accept" per RFC 7231 §5.3.1. Default q is 1 when absent.

CodeRabbit on the same line also pointed out that `startsWith("text/event-stream")` mis-matches structured-suffix media types like `text/event-stream+json` — the new implementation does an exact `===` match after lowercasing/trimming, so that case is also fixed.

Regression tests added in `backend/src/utils/api-helpers.test.ts`.

File: backend/src/utils/api-helpers.ts [hunk 3]
```diff
@@ -46,6 +46,33 @@ export const EXCLUDED_HEADERS = new Set([
...
+      .map((segment) => segment.trim().toLowerCase());
+    if (mediaType !== "text/event-stream") {
+      return false;
+    }
+    const qParam = params.find((p) => p.startsWith("q="));
+    if (qParam === undefined) {
+      return true;
+    }
+    const q = Number(qParam.slice(2));
+    return Number.isFinite(q) && q > 0;
```
[the remark this answers is not in the document]
Comment (pescn) [reply backend/src/utils/api-helpers.ts]: Fixed in 53d6fae — confirmed reproducible against the round-1 implementation:

- `text/event-stream ; q = 0` previously returned `true` (q parsing fell through to the "no q parameter" branch because `"q = 0".startsWith("q=")` is false)
- `text/event-stream ; q = 0.5` returned the right answer for the wrong reason — same fall-through, defaulted to q=1 instead of actually parsing q=0.5

Replaced the `startsWith("q=")` check with an `indexOf("=")` + slice + trim pass that parses name and value separately, so whitespace around the `=` no longer hides `q`. RFC 7230 §3.2.6 permits this whitespace in `parameter = token "=" ...`, so strict clients can legitimately produce these forms.

Regression tests added (12 → 15 cases) covering whitespace `q=0`, whitespace `q=0.5`, and the empty `q=` value edge case.
